atox/src/main/kotlin/ToxService.kt

```diff
@@ -124,6 +132,24 @@ class ToxService : LifecycleService() {
                 }
         }
 
+        friendRequestManager.getAll()
+            .filterNotNull()
+            .asLiveData()
```

Comment:
`LiveData` only runs in the main/gui thread, so use `.collect` on the `Flow` instead of `.observe` on the `LiveData`. You can do what we do above with `lifecycleScope.launch` to get it to automatically unsubscribe when the ToxService is meant to die.

Reply:
Didn't work, when I tried it. Then I changed to LiveData and it worked.

Reply:
It's what we do right above, and that works. Surely it must work here too? Did you not get any callbacks, or what was the problem?

Reply:
I forgot, there was a different issue in the beginning and I noticed it just after I changed to LiveData.